src/core/ext/transport/inproc/inproc_transport.c

```diff
@@ -72,6 +72,7 @@ typedef struct sb_list_entry {
 typedef struct {
   grpc_byte_stream base;
   sb_list_entry *le;
+  grpc_error *shutdown_error;
```

Comment:
I thought about this, but I suspect that there will be implementations (such as the one in chttp2) that will want to handle shutdown differently.